Add TAG S&P self-review questionnaire. (#14)
WICG/CHIPS: TAG-S&P-questionnaire.md

### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?

[Yes](https://github.com/WICG/CHIPS/blob/main/README.md#security-and-privacy-considerations).

### 2.16. Does this specification allow downgrading default security characteristics?

No.

## 3. Threat Models

### 3.1. Passive Network Attackers

The Partitioned attribute can only be used for cookies being sent over secure connections, so the cookie strings are not visible to passive network attackers.

### 3.2. Active Network Attackers

See 3.1.

### 3.3. Same-Origin Policy Violations

None.

### 3.4. Third-Party Tracking

Partitioning cross-site cookies by top-level site is meant to reduce (or eliminate) third parties' ability to use cookies as a cross-site identifier.
The goal of this proposal is for cross-site contexts to be able to observe a user's activity only within a single top-level site.

### 3.5. Legitimate Misuse

A malicious third party may wish to use partitioned cookies to store a global user identifier that persists across multiple top-level domains.

> This identifier could be repeatedly derived on each top-level site a user visits using some side channel.

Reviewer: "... using a side channel, or fingerprinting techniques".

Author: Done.

> Afterwards the third-party can store this identifier in a partitioned cookie so that when the user returns to each top-level site with the third-party embed, they no longer need to use the side channel to derive the cross-site identifier again.

Reviewer: Might be worth adding that the same capability can be achieved by third-party scripts with access to the first-party cookie jar.

Author: Done.

# Self-Review Questionnaire: Security and Privacy

Questions from w3.org/TR/security-privacy-questionnaire

## 2. Questions to Consider

### 2.1. What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?

If unpartitioned third-party cookies are still in use, this feature does not expose any additional information that is not already available if sites were already using an unpartitioned cross-site cookie.

> If unpartitioned third-party cookies are blocked, this provides sites in third-party contexts the ability to use HTTP cookies that are scoped to a single top-level site.

Reviewer 2: "the ability to use HTTP cookies" doesn't really answer the question, unless you already know what information a cookie exposes. Maybe:

```suggestion
If unpartitioned third-party cookies are blocked, this provides sites in third-party contexts the ability to save information across visits, within the scope of a single top-level site.
```

We should also answer the "for what purposes" question. It's to track sessions, right?

Author: Done.

### 2.2. Is this specification exposing the minimum amount of information necessary to power the feature?

> Yes. This proposal gives the ability for third-party contexts the ability to use a familiar HTTP state primitive for use cases restricted to a users' activity in only the current top-level site.

Reviewer 2: Would it make sense to mention that HTTP State Tokens expose a bit less information, but that they seem to require too much effort from site operators to adopt?

Author: How do they reveal less information? Is it because they are client generated and their value is not set by the site?

Reviewer 2: Yeah, so there's less space total and no worry about the site putting anything sensitive there.

Author: Got it. Ok, added something about HST's here.

### 2.3. How does this specification deal with personal information or personally-identifiable information or information derived thereof?

> This proposal requires that cookies which use the Partitioned attribute also have the __Host- prefix.

Reviewer 2: Start by pointing out that a server might put PII into a cookie, or attach a unique-ID-in-a-cookie to PII on the server, so it's important to protect that from network eavesdroppers. 

This is probably also a good place to brag that partitioned cookies can't share that PII between different top-level sites.

Author: > Start by pointing out that a server might put PII into a cookie, or attach a unique-ID-in-a-cookie to PII on the server, so it's important to protect that from network eavesdroppers.

Done.

> This is probably also a good place to brag that partitioned cookies can't share that PII between different top-level sites.

Done.

This prefix requires cookies be hostname scoped and can only be used in [secure contexts](https://www.w3.org/TR/secure-contexts/).
The latter requirement protects these cookies from sending PII in plaintext between clients and servers.

### 2.4. How does this specification deal with sensitive information?

See 2.3.

### 2.5. Does this specification introduce new state for an origin that persists across browsing sessions?

This proposal extends an existing state mechanism that persists across browsing sessions.
It imposes different types of size limitations on these types of cookies in order to prevent them from being abused as a cross-site tracking vector.

> For more information for this type of attack, see this section of the explainer.

Reviewer 2: Did you lose a link to an explainer section?

Author: Whoops, fixed.

### 2.6. What information from the underlying platform, e.g. configuration data, is exposed by this specification to an origin?

This feature should not reveal any additional information about the user's underlying platform.

### 2.7. Does this specification allow an origin access to sensors on a user’s device

No.

> ### 2.8. What data does thisi UI specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.

Reviewer 2: ```suggestion
### 2.8. What data do the features in this specification expose to an origin? Please also document what data is identical to data exposed by other features, in the same or different contexts.
```

Author: Done.

### 2.13. How does this specification distinguish between behavior in first-party and third-party contexts?

The Partitioned attribute is designed for use in third-party contexts, but that does not mean that they cannot be used in first-party contexts.
When used in first-party contexts, their semantics are identical to that of first-party cookies.

### 2.14. How does this specification work in the context of a user agent’s Private \ Browsing or "incognito" mode?

Like all other cookies in private browsers, partitioned cookies should be cleared when a private browsing session ends.
Some browser private settings also block the use of cross-site cookies.
User agents may wish to implement exceptions for partitioned cookies for these cross-site cookie restrictions for private browsing mode.

### 2.15. Does this specification have a "Security Considerations" and "Privacy Considerations" section?

[Yes](https://github.com/WICG/CHIPS/blob/main/README.md#security-and-privacy-considerations).

### 2.16. Does this specification allow downgrading default security characteristics?

No.

> ## 3. Threat Models

Reviewer 2: We usually don't include comments on the threat models in the S&P questionnaire answers. You have good answers here, though, so it makes sense to keep them.

Author: Ack'd.

### 3.1. Passive Network Attackers

The Partitioned attribute can only be used for cookies being sent over secure connections, so the cookie strings are not visible to passive network attackers.

### 3.2. Active Network Attackers

See 3.1.

### 3.3. Same-Origin Policy Violations

> None.

Reviewer 2: `__Host` cookies still violate the same-origin policy by allowing different ports to access the same cookie, right?

Author: Good point. Changed.

### 3.4. Third-Party Tracking

Partitioning cross-site cookies by top-level site is meant to reduce (or eliminate) third parties' ability to use cookies as a cross-site identifier.

> The goal of this proposal is for cross-site contexts to be able to observe a user's activity only within a single top-level site.

Reviewer 2: ```suggestion
The goal of this proposal is for cross-site contexts to be able to associate a user's activity only within a single top-level site.
```
?

Author: Done.

### 3.5. Legitimate Misuse

> A malicious third party may wish to use partitioned cookies to store a global user identifier that persists across multiple top-level domains.

Reviewer 2: I'd remove "malicious" here, and mention [UID2.0](https://www.thetradedesk.com/us/knowledge-center/what-the-tech-is-unified-id-2-0) or another similar global ID system. Different readers will have different understandings of whether that's a malicious use. I think we'll still get pushback on whether it's "misuse" at all, but I think it belongs best in this section, and that it's worth mentioning.

Author: > I'd remove "malicious" here

Done.

> mention UID2.0 or another similar global ID system.

Done, though not sure if how I am mentioning it is in line with what you're thinking.

This identifier could be repeatedly derived on each top-level site a user visits using some side channel or fingerprinting techniques.
Afterwards the third-party can store this identifier in a partitioned cookie so that when the user returns to each top-level site with the third-party embed, they no longer need to use the side channel to derive the cross-site identifier again.

> Though partitioned cookies could introduce a new place for third parties to store a derived cross-site identifier, a third-party script running in a top-level context could store still store a cross-site identifier in the top-level site's cookie jar.

Reviewer 2: ```suggestion
Though partitioned cookies could introduce a new place for third parties to store a derived cross-site identifier, a third-party script running in a top-level context could still store a cross-site identifier in the top-level site's cookie jar.
```

Author: Done.

### 3.4. Third-Party Tracking

Partitioning cross-site cookies by top-level site is meant to reduce (or eliminate) third parties' ability to use cookies as a cross-site identifier.
The goal of this proposal is for cross-site contexts to be able to associate a user's activity only within a single top-level site.

### 3.5. Legitimate Misuse

A third party may wish to use partitioned cookies to store a global user identifier that persists across multiple top-level domains.
This identifier could be repeatedly derived on each top-level site a user visits using some side channel or fingerprinting techniques.
Afterwards the third-party can store this identifier in a partitioned cookie so that when the user returns to each top-level site with the third-party embed, they no longer need to use the side channel to derive the cross-site identifier again.

Though partitioned cookies could introduce a new place for third parties to store a derived cross-site identifier, a third-party script running in a top-level context could still store a cross-site identifier in the top-level site's cookie jar.

> Also, there are proposals for identifiers (e.g. [UID2.0](https://www.thetradedesk.com/us/knowledge-center/what-the-tech-is-unified-id-2-0)) which may disincentivize third parties from using partitioned cookies as a replacement to third-party cookies for cross-site tracking.

Reviewer 2: I was thinking that UID2.0 is an example of an identifier that's "repeatedly derived on each top-level site a user visits", and which might be conveniently stored in a partitioned cookie, but I guess that standard also describes how to store it in the top-level site's cookie jar, so it's really an example of how the ecosystem has already gone through the whole flow you're describing.

No change needed.

Author: Ack'd
